controller/users.go

```diff
@@ -157,6 +193,137 @@ func (c *UsersController) getKeycloakProfileInformation(ctx context.Context, tok
+			isValid := isEmailValid(*updatedEmail)
+			if !isValid {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to email for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+				return ctx.BadRequest(jerrors)
+			}
+			isUnique, err := isEmailUnique(appl, *updatedEmail, *user)
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("error updating identitity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+			}
+			if !isUnique {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("email address: %s is already in use", *updatedEmail)))
+				return ctx.Conflict(jerrors)
+			}
+			user.Email = *updatedEmail
+		}
+
+		updatedUserName := ctx.Payload.Data.Attributes.Username
+		if updatedUserName != nil && *updatedUserName != identity.Username {
+			isValid := isUsernameValid(*updatedUserName)
+			if !isValid {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to username for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+				return ctx.BadRequest(jerrors)
+			}
+			if identity.RegistrationCompleted {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("username cannot be updated more than once for identity id %s ", *id)))
+				return ctx.Forbidden(jerrors)
+			}
+			isUnique, err := isUsernameUnique(appl, *updatedUserName, *identity)
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("error updating identitity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+			}
+			if !isUnique {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("username : %s is already in use", *updatedUserName)))
+				return ctx.Conflict(jerrors)
+			}
+			identity.Username = *updatedUserName
+
+		}
+
+		updatedRegistratedCompleted := ctx.Payload.Data.Attributes.RegistrationCompleted
+		if updatedRegistratedCompleted != nil {
+			if !*updatedRegistratedCompleted {
+				jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrInvalidRequest(fmt.Sprintf("invalid value assigned to registration_completed for identity with id %s and user with id %s", identity.ID, identity.UserID.UUID)))
+				log.Error(ctx, map[string]interface{}{
+					"registration_completed": *updatedRegistratedCompleted,
+					"user_id":                identity.UserID.UUID,
+					"identity_id":            identity.ID,
+				}, "invalid parameter assignment")
+
+				return ctx.BadRequest(jerrors)
+			}
+			identity.RegistrationCompleted = true
+		}
+
+		updatedBio := ctx.Payload.Data.Attributes.Bio
+		if updatedBio != nil && *updatedBio != user.Bio {
+			user.Bio = *updatedBio
+		}
+		updatedFullName := ctx.Payload.Data.Attributes.FullName
+		if updatedFullName != nil && *updatedFullName != user.FullName {
```

Comment:
Don't modify the fields which are coming from Auth. Auth will modify the original full name update request and pass it along.

Reply:
https://github.com/fabric8-services/fabric8-wit/pull/1637/commits/fccb1b01c0b946bfeb2d8be8e4e948f06f7ac45f#diff-da062b312519ee2a61dd669f1ec964d4